Rules.modular

```diff
@@ -138,26 +146,25 @@ $(tmpdir)/global_bools.conf: M4PARAM += -D self_contained_policy
 $(tmpdir)/global_bools.conf: $(m4support) $(tmpdir)/generated_definitions.conf $(globalbool) $(globaltun)
 	$(verbose) $(M4) $(M4PARAM) $^ > $@
 
-$(tmpdir)/all_interfaces.conf: $(m4support) $(all_interfaces) $(m4iferror)
-	@test -d $(tmpdir) || mkdir -p $(tmpdir)
+$(tmpdir)/all_interfaces.conf: | $(tmpdir)
+$(tmpdir)/all_interfaces.conf: $(tmpdir)/m4support.m4f $(all_interfaces) $(m4iferror)
 	@cat $(m4divert) > $@
-	$(verbose) $(M4) $^ > $(tmpdir)/$(@F).tmp
-	$(verbose) $(SED) -e s/dollarsstar/\$$\*/g $(tmpdir)/$(@F).tmp >> $@
+	$(verbose) $(M4) $(M4PARAM) -R $^ | $(SED) -e s/dollarsstar/\$$\*/g >> $@
```

Comment:
Hello, quick comment for the change "use pipe instead of tmp file": if I understand correctly how `make` and `sh` usually work, when a pipe is used to run `cmd1 | cmd2` and `cmd1` fails but `cmd2` succeeds, the whole command is seen as "success" (and `set -o pipefail` in bash prevents such an issue).

Here if the `m4` invocation fails, the build will no longer fail. This seems to be a regression. Is my interpretation correct? I might have missed where the "pipefail" option configured in this Makefile.

Reply:
You are right
```
$ cat Makefoo 
a:
	m4 <<< "m4exit(\`1')" | sed -e s/1/2/
b:
	m4 <<< "m4exit(\`1')" > tmp1
	sed -e s/1/2/ tmp1
$ make -f Makefoo 
m4 <<< "m4exit(\`1')" | sed -e s/1/2/
$ echo $?
0
$ make -f Makefoo b
m4 <<< "m4exit(\`1')" > tmp1
make: *** [Makefoo:5: b] Error 1
$ echo $?
2
```

Reply:
I'd prefer to not use any bashisms and stay POSIX compliant.

Reply:
I'd also prefer to have intermediate files since there is so much processing. If something goes wrong it will be harder to find. Build performance isn't a concern for refpolicy; correctness is.

Reply:
Ok, testfiles without <<< bashism
```
$ cat testmake
# .SHELLFLAGS := -c -o pipefail

a:
	echo "m4exit(\`1')" | m4 - | sed -e s/1/2/
b:
	echo "m4exit(\`1')" | m4 - > tmp1
	sed -e s/1/2/ tmp1
$ cat testmake-pipefail 
.SHELLFLAGS := -c -o pipefail

a:
	echo "m4exit(\`1')" | m4 - | sed -e s/1/2/
b:
	echo "m4exit(\`1')" | m4 - > tmp1
	sed -e s/1/2/ tmp1
```

Now without -o pipefail, using pipe does not fail:
```
$ make -f testmake -k a b
echo "m4exit(\`1')" | m4 - | sed -e s/1/2/
echo "m4exit(\`1')" | m4 - > tmp1
make: *** [testmake:6: b] Error 1
```

With -o pipefail, both examples fail:
```
$ make -f testmake-pipefail -k a b
echo "m4exit(\`1')" | m4 - | sed -e s/1/2/
make: *** [testmake-pipefail:4: a] Error 1
echo "m4exit(\`1')" | m4 - > tmp1
make: *** [testmake-pipefail:6: b] Error 1
```

As far as I can see, any current pipe with any possibility of failure is not okay without -o pipefail.

Reply:
I'm fine with changing this to a pipe as long as the suggested `.SHELLFLAGS` are added.